src/game_launcher.cpp

```diff
@@ -843,16 +846,20 @@ void game_launcher::start_wesnothd()
 	SetEnvironmentVariableA("SDL_STDIO_REDIRECT", "0");
 #endif
 	LOG_GENERAL << "Starting wesnothd: "<< command;
-	if (std::system(command.c_str()) == 0) {
-		// Give server a moment to start up
+	try
+	{
+		bp::spawn(command);
```

Comment:
Should be something like:
```cpp
bp::spawn(wesnothd_program, "-c", config, "-d", "-t", "2", "-T", "5")
```

I guess omit the `-d` on Windows? I don't really get why though. And it looks like Windows is using some magic feature to set the console titlebar, which would be lost with the switch to Boost.Process, so I'd suggest making wesnothd set its own title instead.

But basically, the change made here does _not_ address the CodeQL issue.

Reply:
For what it's worth, setting the title within wesnothd is about this simple:

```cpp
std::string title = "wesnoth server";
#ifdef WIN32
	SetConsoleTitleA(title.c_str());
#else
	cout << "\033]0;" << title << "\007";
#endif
```

Reply:
Where would this go exactly? Just in main()?

Reply:
Yeah, after argument parsing.